File: tilingol.py
#!/usr/bin/env python

import sys
print len(sys.argv)

if len(sys.argv) < 5:
    print "Usage tilingol.py FREQUENCY_WINDOW MOVING_AVERAGE_WINDOW PEAK_THRESHOLD KEYWORDS [LANGS]"
    exit()

TRACK    = sys.argv[4]
LANGUAGE = sys.argv[5] if len(sys.argv) == 6 else ""

from goalstream import GoalStream

txtfile = open("config/credentials.txt", "r")
credentials = txtfile.readlines()
txtfile.close()

APP_KEY = credentials[0].strip()
APP_SECRET = credentials[1].strip()
OAUTH_TOKEN = credentials[2].strip()
OAUTH_TOKEN_SECRET = credentials[3].strip()

print "Howdy! I'm tracking:"
print "  TRACK     = " + TRACK
print "  LANGUAGE  = " + LANGUAGE
print sys.argv[1] + " - " + sys.argv[2] + " - " + sys.argv[3] + "\n"

stream = GoalStream(APP_KEY, APP_SECRET, OAUTH_TOKEN, OAUTH_TOKEN_SECRET)
stream.configure(int(sys.argv[1]), int(sys.argv[2]), float(sys.argv[3]), True)
stream.statuses.filter(track=TRACK, language=LANGUAGE)

File: tilingol.rb
#!/usr/bin/env ruby

require "rubygems"
require "tweetstream"
require "yaml"
require "json"
require "uri"
require_relative "./peak_detection"
require_relative "./stdout_bell"
require_relative "./jingle_bell"

# CONFIGURATION
@keywords      = "gol,gool,goool,golaço" 
@languages     = ""
@peak_detector = Tilingol::PeakDetection.new(1,10,1.5, true) #frequency window, moving average window, peak threshold
@bell          = Tilingol::StdoutBell.new # or JingleBell.new | use the class according with the bell wanted
@magic_word    = "toqueosinopequenino" # tweet with this word + a tracker keyword to ring the bell anyway
# END OF CONFIGURATION

unless ENV['TW_CONSUMER_KEY']
  # Will load credentials.yml, make sure it's there
  # This must to be used only for development environment
  @oauth = YAML.load_file(File.expand_path("./config/credentials.yml"))

  ENV['TW_CONSUMER_KEY'] = @oauth["consumer_key"]
  ENV['TW_CONSUMER_SECRET'] = @oauth["consumer_secret"]
  ENV['TW_OAUTH_TOKEN'] = @oauth["access_token"]
  ENV['TW_OAUTH_TOKEN_SECRET'] = @oauth["access_token_secret"]
end

# configure tweetstream instance
TweetStream.configure do |config|
  config.consumer_key       = ENV['TW_CONSUMER_KEY']
  config.consumer_secret    = ENV['TW_CONSUMER_SECRET']
  config.oauth_token        = ENV['TW_OAUTH_TOKEN']
  config.oauth_token_secret = ENV['TW_OAUTH_TOKEN_SECRET']
  config.auth_method = :oauth
end

@client = TweetStream::Client.new

@client.on_error do |message|
  puts "ERROR: #{message}"
end
@client.on_enhance_your_calm do
  puts "Calm down"
end
@client.on_limit do |skip_count|
  puts "You lost #{skip_count} tweets"
end

puts "Starting to track: #{@keywords}...\nLanguages: #{@languages}"
@client.filter(:track => @keywords, :language => @languages) do |status| 
  @peak_detector.collect_frequency

  @bell.ring! if @peak_detector.is_this_a_peak? || status.text.index(@magic_word)

  #puts status.text
end

